libcore/FileUtils.vala

```diff
@@ -881,6 +881,52 @@ namespace PF.FileUtils {
         }
     }
 
+    public string get_afp_target_uri (string standard_target_uri, string uri) {
+        /* For afp:// addresses the standard target uri has the user name stripped - we need to replace it */
+        if (Uri.parse_scheme (uri) == "afp") {
+            string origin_host, origin_filename;
+            if (get_afp_user_server_and_filename (uri, out origin_filename, out origin_host)) {
+                string target_host, target_filename;
+                if (get_afp_user_server_and_filename (standard_target_uri, out target_filename, out target_host)) {
+                    return "afp://" + Path.build_path (Path.DIR_SEPARATOR_S, origin_host, target_filename);
+                }
+            }
+        } else if (Uri.parse_scheme (standard_target_uri) == "afp") {
+            /* Probably clicked on an AFP server in network:// view which gives weird gvfs uri like:
+             * network:///dnssd-domain-<host>._afpovertcp._tcp.  We have to get any missing username from
+             * the Environment */
+            string target_host, target_filename;
+            if (get_afp_user_server_and_filename (standard_target_uri, out target_filename, out target_host)) {
+                if (!target_host.contains ("@")) {
+                    /* Cannot get username from origin uri so just use Environment user name */
+                    target_host = GLib.Environment.get_user_name () + "@" + target_host.dup ();
```

Comment:
Hrm is that reasonable to assume the computer user name is the same as the afp user name? Maybe in this case where we can't get the username we should be prompting. Or will this fail to prompting for the user name if the uri turns out to be invalid?

Reply:
This gets called every time the file updates and we do not want to repeatedly throw dialogs.  I could do some more testing of error situations - I haven't worked on this since reinstalling so I need to set things up again.

Reply:
On further thoughts, if this work around fails it will just cause the normal system prompt to show instead of automatically connecting (as now) so I do not think there is any point in throwing our own dialog.

Reply:
On even further thought, I think it best to omit the username altogether in this circumstance - this triggers a system dialog requesting the name as well as the password.